src/snowflake/cli/plugins/nativeapp/manager.py

```diff
@@ -218,10 +218,34 @@ def stage_schema(self) -> Optional[str]:
     def package_warehouse(self) -> Optional[str]:
         return self.na_project.package_warehouse
 
+    def use_package_warehouse(self) -> str:
+        if self.package_warehouse:
+            return self.package_warehouse
+        raise ClickException(
+            dedent(
+                f"""\
+            Application package warehouse cannot be empty.
+            Please provide a value for it in your connection information or your project definition file.
+            """
+            )
+        )
+
     @property
     def application_warehouse(self) -> Optional[str]:
         return self.na_project.application_warehouse
 
+    def use_application_warehouse(self) -> str:
+        if self.application_warehouse:
+            return self.application_warehouse
```

Comment:
Rather than return a string, I would expect this to be:
```
if self.application_warehouse:
  with self.use_warehouse(self.application_warehouse):
    yield
else:
  raise ...
```